packets.go

```diff
@@ -83,7 +83,15 @@ func (mc *mysqlConn) readPacket() ([]byte, error) {
 }
 
 // Write packet buffer 'data'
-func (mc *mysqlConn) writePacket(data []byte) error {
+func (mc *mysqlConn) writePacket(ctx mysqlContext, data []byte) error {
+	if ctx == nil {
+		panic("context cannot be nil")
+	}
+	ctxDeadline, isCtxDeadlineSet := ctx.Deadline()
+	if isCtxDeadlineSet && !ctxDeadline.After(time.Now()) {
+		return errors.New("timeout")
```

Comment:
Could we use `context. DeadlineExceeded`?
https://golang.org/pkg/context/#pkg-variables